pages/_app.js

```diff
@@ -1,10 +1,18 @@
+import { useState } from "react";
 import GlobalStyle from "../styles";
 
 export default function App({ Component, pageProps }) {
-	return (
-		<>
-			<GlobalStyle />
-			<Component {...pageProps} />
-		</>
-	);
+  //Hier States
+  const [productToBuy, setProduct] = useState();
+
+  return (
+    <>
+      <GlobalStyle />
+      <Component
+        {...pageProps}
+        setProduct={setProduct}
```

Comment:
You are using the props directly for the `Component`. Depending on the complexity of your app you might run into prop drilling if you need to pass the props through multiple components. I remember that there is a React context solution so you could consider to check this out instead.